src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1803,8 +1800,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			title: nls.localize('changeTerminalIcon', "Change Icon"),
 			matchOnDescription: true
 		});
-		if (result) {
-			this.shellLaunchConfig.icon = result.description;
+		if (result && result.description) {
+			this.shellLaunchConfig.icon = { id: result.description };
```

Comment:
nit: We could avoid creating a new object here:

```suggestion
			this.shellLaunchConfig.icon = iconRegistry.get(result.description);
```